benchmarks/map-reduce/driver/main.py

```diff
@@ -20,253 +20,192 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-from __future__ import print_function
-
-import sys
 import os
+import sys
 
-# adding python tracing sources to the system path
-sys.path.insert(0, os.getcwd() + '/../proto/')
-sys.path.insert(0, os.getcwd() + '/../../../utils/tracing/python')
-import tracing
-import helloworld_pb2_grpc
-import helloworld_pb2
-import mapreduce_pb2_grpc
-import mapreduce_pb2
-import destination as XDTdst
-import source as XDTsrc
-import utils as XDTutil
-
-import grpc
-from grpc_reflection.v1alpha import reflection
-import argparse
-import boto3
+from concurrent import futures
 import logging as log
-import socket
-import json
-import resource
-from io import StringIO ## for Python 3
-import time
-from joblib import Parallel, delayed
 
-import pickle
+import tracing
+from driver import DriveFunction
+
+LAMBDA = os.environ.get('IS_LAMBDA', 'no').lower() in ['true', 'yes', '1']
+TRACE = os.environ.get('ENABLE_TRACING', 'no').lower() in ['true', 'yes', '1', 'on']
+
+if LAMBDA:
+    import boto3
+    import json
+
+if not LAMBDA:
+    import grpc
+    from grpc_reflection.v1alpha import reflection
+    import argparse
+
+    import helloworld_pb2_grpc
+    import helloworld_pb2
+    import mapreduce_pb2_grpc
+    import mapreduce_pb2
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-dockerCompose", "--dockerCompose",
+        dest="dockerCompose", default=False, help="Env docker compose")
+    parser.add_argument("-mAddr", "--mAddr", dest="mAddr",
+        default="mapper.default.192.168.1.240.sslip.io:80",
+        help="trainer address")
+    parser.add_argument("-rAddr", "--rAddr", dest="rAddr",
+        default="reducer.default.192.168.1.240.sslip.io:80",
+        help="reducer address")
+    parser.add_argument("-sp", "--sp", dest="sp", default="80",
+        help="serve port")
+    parser.add_argument("-zipkin", "--zipkin", dest="zipkinURL",
+        default="http://zipkin.istio-system.svc.cluster.local:9411/api/v2/spans",
+        help="Zipkin endpoint url")
+
+    args = parser.parse_args()
+
+if TRACE:
+    # adding python tracing sources to the system path
+    sys.path.insert(0, os.getcwd() + '/../proto/')
+    sys.path.insert(0, os.getcwd() + '/../../../utils/tracing/python')
+
+    if tracing.IsTracingEnabled():
```

Comment:
I think we need to import tracing after adding to the sys.path

Reply:
Oh okay, now I understand what those `sys.path.insert`s were!
In the [Dockerfile](https://github.com/vhive-serverless/vSwarm/blob/main/benchmarks/map-reduce/Dockerfile#L31) `tracing.py` is copied alongside the file containing the function code, so `import tracing` works fine.

Should I remove the two lines? I think removing them entirely would be better than placing them above the import statement since they are not really required.

Reply:
Perfect. Thanks :smile: